### PR TITLE
internal: refactoring to prepare for roles

### PR DESCRIPTION
**Pull request by oleg-jukovec**
This patchset contains an internal refactoring to implement roles for Tarantool 3.0. It splits roles for Tarantool Cartridge to a logical blocks that could be reused to build Tarantool 3.0 roles.

Part of #68

### CODE REVIEW

File: sharded_queue/api.lua [hunk 1]
```diff
@@ -148,13 +144,19 @@ function sharded_tube.take(self, timeout, options)
     while take_timeout ~= 0 do
         local begin = time.cur()
 
-        local storages = cartridge_rpc.get_candidates(
-            'sharded_queue.storage',
```
[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply sharded_queue/api.lua:152]: Yes, for the router's cluster. But it should not cause any problem in the current sharded-queue installations.

File: sharded_queue/storage/drivers.lua [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    driver_name = driver_name or DEFAULT_DRIVER
+
+    if queue_drivers[driver_name] == nil then
+        queue_drivers[driver_name] = require(driver_name)
+    end
+    return queue_drivers[driver_name]
+end
+
+return {
+    get = get_driver
```
[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply sharded_queue/storage/drivers.lua]: Updated.

File: sharded_queue/storage/tubes.lua [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    return new, old
+end
+
+local mt = {
+    __index = {
+        call = call,
+        get_options = get_options,
+        map = map,
+        update = update,
+    }
```
[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply sharded_queue/storage/tubes.lua]: Updated.

File: sharded_queue/api.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (DifferentialOrange) [reply sharded_queue/api.lua:152]: Let's write about it in README, just in case.

[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply sharded_queue/api.lua:152]: The README.md is updated.

**Review by DifferentialOrange — APPROVED** [review]

File: test/storage_test.lua [hunk 4]
```diff
@@ -17,7 +16,7 @@ g.test_storage_methods = function()
     local ro = g.storage_ro:eval("return box.cfg.read_only")
     t.assert_equals(ro, true)
 
-    for _,method  in pairs(storage.__private.methods) do
+    for _,method  in pairs(methods.get_list()) do
```
Comment (DerekBum) [inline test/storage_test.lua:19]:
```suggestion
    for _, method in pairs(methods.get_list()) do
```

File: sharded_queue/router/tube.lua [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+            options,
+            call_timeout
+        )
+
+        if ret ~= nil and ok then
+            return ret
+        end
+
+        local duration = time.cur() - begin
+        take_timeout = take_timeout > duration and take_timeout -  duration or 0
```
Comment (DerekBum) [inline sharded_queue/router/tube.lua:37]:
```suggestion
        take_timeout = take_timeout > duration and take_timeout - duration or 0
```

File: sharded_queue/router/tube.lua [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+    options.bucket_id = bucket_id
+    options.bucket_count = bucket_count
+
+    options.extra = {
+        log_request = utils.normalize.log_request(options.log_request) or self.log_request,
+    }
+
+    local task, err = vshard.router.call(bucket_id,
+        'write', 'tube_put', { options })
+    -- re-raise storage errors
```
Comment (DerekBum) [inline sharded_queue/router/tube.lua:62]:
```suggestion
    -- Re-raise storage errors.
```

Up to you.

File: sharded_queue/router/tube.lua [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+    options.task_id = task_id
+
+    options.extra = {
+        log_request = utils.normalize.log_request(options.log_request) or self.log_request,
+    }
+
+    local task, err = vshard.router.call(bucket_id, 'write', 'tube_delete', {
+        options
+    })
+    -- re-raise storage errors
```
Comment (DerekBum) [inline sharded_queue/router/tube.lua:168]:
```suggestion
    -- Re-raise storage errors.
``` 
And so on. There is a bunch of copies of this comment. Up to you.

**Review by DerekBum — APPROVED** [review]
Lgtm! Just a few little comments.

File: test/storage_test.lua (hunk 4 above)
Comment (oleg-jukovec) [reply test/storage_test.lua]: Thank you, updated.

File: sharded_queue/router/tube.lua (hunk 6 above)
Comment (oleg-jukovec) [reply sharded_queue/router/tube.lua]: Thank you, updated.

File: sharded_queue/router/tube.lua (hunk 7 above)
Comment (oleg-jukovec) [reply sharded_queue/router/tube.lua]: Thank you, updated.

**Pull request merged**